packages/data/src/components/use-dispatch/test/use-dispatch.js

```diff
@@ -11,165 +12,171 @@ import createReduxStore from '../../../redux-store';
 import { createRegistry } from '../../../registry';
 import { RegistryProvider } from '../../registry-provider';
 
+jest.useRealTimers();
+
 describe( 'useDispatch', () => {
+	const counterStore = {
+		reducer: ( state = 0, action ) => {
+			if ( action.type === 'INC' ) {
+				return state + 1;
+			}
+
+			return state;
+		},
+		actions: {
+			inc: () => ( { type: 'INC' } ),
+		},
+		selectors: {
+			get: ( state ) => state,
+		},
+	};
+
 	let registry;
 	beforeEach( () => {
 		registry = createRegistry();
 	} );
 
-	it( 'returns dispatch function from store with no store name provided', () => {
-		registry.registerStore( 'demoStore', {
-			reducer: ( state ) => state,
-			actions: {
-				foo: () => 'bar',
-			},
-		} );
+	it( 'returns dispatch function from store with no store name provided', async () => {
+		const user = userEvent.setup();
+		registry.registerStore( 'demo', counterStore );
+
 		const TestComponent = () => {
-			return <div></div>;
-		};
-		const Component = () => {
 			const dispatch = useDispatch();
-			return <TestComponent dispatch={ dispatch } />;
+			return <button onClick={ () => dispatch( 'demo' ).inc() } />;
 		};
 
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create(
-				<RegistryProvider value={ registry }>
-					<Component />
-				</RegistryProvider>
-			);
-		} );
+		render(
+			<RegistryProvider value={ registry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
 
-		const testInstance = testRenderer.root;
+		await user.click( screen.getByRole( 'button' ) );
 
-		expect( testInstance.findByType( TestComponent ).props.dispatch ).toBe(
-			registry.dispatch
-		);
+		expect( registry.select( 'demo' ).get() ).toBe( 1 );
 	} );
-	it( 'returns expected action creators from store for given storeName', () => {
-		const noop = () => ( { type: '__INERT__' } );
-		const testAction = jest.fn().mockImplementation( noop );
-		const store = createReduxStore( 'demoStore', {
-			reducer: ( state ) => state,
-			actions: {
-				foo: testAction,
-			},
-		} );
+
+	it( 'returns expected action creators from store for given storeName', async () => {
```

Comment:
The descriptions of the two tests, one for "store name," other for "store descriptor," are flipped. The "store name" one actually tests a descriptor and vice versa.

Reply:
Thanks for catching this. I will do a quick follow-up tomorrow.